src/compiler/builder.ts

```diff
@@ -549,13 +547,11 @@ namespace ts {
         const seenFileAndExportsOfFile = new Set<string>();
         // Go through exported modules from cache first
         // If exported modules has path, all files referencing file exported from are affected
-        if (forEachEntry(state.currentAffectedFilesExportedModulesMap, (exportedModules, exportedFromPath) =>
+        forEachEntry(state.currentAffectedFilesExportedModulesMap, (exportedModules, exportedFromPath) =>

```

Comment:
If this queues files , shouldnt we stop here and not do next step.. its duplicated work?

Reply:
I'm not sure what it _should_ do, but I believe this is just a clearer way to write what it already _does_ do.  I don't understand the code very well and may have missed a case where it does return true and exit early.

Reply:
Specifically, I'm less confident in the second commit than the first.  AFAICT, it could only return true if a recursive call to itself returned true and all of the recursive base cases returned false.